### ISSUE TITLE
Authentication is sloppy

### ISSUE THREAD

**Issue by mosajjal** [qwj/python-proxy#23]
Hi, As you know, I've been using `pproxy` for a while now and I just found out there's a problem with the Authentication scheme. If I Authenticate once with the proxy, my public IP address remains authenticated with the server forever. After the first request, the auth becomes optional for that IP. Is that behavior intended? Because it would pose a big security vulnerability in public networks IMO

**Comment by qwj** [comment]
It is the designed feature. The authenticated time is defined as AUTH_TIME.
The reason for this design is that some OS (MacOS) doesn't send authentication after successfully authenticated once. Maybe we can set AUTH_TIME lower to get higher security level.

```
AUTH_TIME = 86400 * 30
class AuthTable(object):
    _auth = {}
    def __init__(self, remote_ip):
        self.remote_ip = remote_ip
    def authed(self):
        return time.time() - self._auth.get(self.remote_ip, 0) <= AUTH_TIME
    def set_authed(self):
        self._auth[self.remote_ip] = time.time()
```

**Comment by mosajjal** [comment]
I see. Maybe we can add an option to override the multiplier (86400) from the arguments. I think Chrome/Firefox extensions, SwitchyOmega for example, don't depend on the OS to send auth requests so for MAC users it would only be a problem if they set the proxy system-wide.

**Comment by qwj** [comment]
Hi, v2.0.7 added a new parameter --auth to fix this issue:
--auth AUTHTIME   re-auth time interval for same ip (default: 86400*30)

**Comment by mosajjal** [comment]
you're the man :+1: closing it since it's fixed in 644b20c36b860f318d68a340efbafb6245b85c78
